Sming/Libraries/ArduCAM/ArduCAM.cpp

```diff
@@ -0,0 +1,1028 @@
+	Wire.write(regID & 0x00FF); 	
+	Wire.endTransmission();
+	
+	Wire.requestFrom((sensor_addr >> 1),1);
+	if(Wire.available())
+		*regDat = Wire.read(); 	
+
+	delay(1);
+  return(1);
+}
+
+//I2C Write 8bit address, 16bit data
+byte ArduCAM::wrSensorReg8_16(int regID, int regDat)
+{
+	Wire.beginTransmission(sensor_addr >> 1);
+	Wire.write(regID & 0x00FF); 	
+
+	Wire.write(regDat >> 8);            // sends data byte, MSB first
+  	Wire.write(regDat & 0x00FF);  	
+	
+	if(Wire.endTransmission())
+	{
+		return 0; 
+	}
+
+	delay(1);
+  	return(1);
+}
+
+//I2C Read 8bit address, 16bit data
+byte ArduCAM::rdSensorReg8_16(uint8_t regID, uint16_t* regDat)
+{
+	uint8_t temp;
+	Wire.beginTransmission(sensor_addr >> 1);
+	Wire.write(regID); 	
+	Wire.endTransmission();
+	
+	Wire.requestFrom((sensor_addr >> 1),2);
+	if(Wire.available())
+	{
+		temp = Wire.read();
+		*regDat = (temp << 8) | Wire.read();
+	}
+
+	delay(1);
+  	return(1);
+}
+
+//I2C Write 16bit address, 8bit data
+byte ArduCAM::wrSensorReg16_8(int regID, int regDat)
+{
+	Wire.beginTransmission(sensor_addr >> 1);
+	Wire.write(regID >> 8);            // sends instruction byte, MSB first
+  	Wire.write(regID & 0x00FF); 	
+  	Wire.write(regDat & 0x00FF);  	
+
+	if(Wire.endTransmission())
+	{
+		return 0; 
+	}
+
+	delay(1);
+  return(1);
+}
+
+//I2C Read 16bit address, 8bit data
+byte ArduCAM::rdSensorReg16_8(uint16_t regID, uint8_t* regDat)
+{
+	Wire.beginTransmission(sensor_addr >> 1);
+	Wire.write(regID >> 8);
+	Wire.write(regID & 0x00FF); 	
+	Wire.endTransmission();
+	
+	Wire.requestFrom((sensor_addr >> 1),1);
+	if(Wire.available())
+	{
+		*regDat = Wire.read(); 	
+	}
+	delay(1);
+  return(1);
+}
+
+//I2C Write 16bit address, 16bit data
+byte ArduCAM::wrSensorReg16_16(int regID, int regDat)
+{
+	Wire.beginTransmission(sensor_addr >> 1);
+	
+	Wire.write(regID >> 8);            // sends instruction byte, MSB first
+  	Wire.write(regID & 0x00FF); 	
+
+  	Wire.write(regDat >> 8);            // sends data byte, MSB first
+  	Wire.write(regDat & 0x00FF);  	
+
+	if(Wire.endTransmission())
+	{
+		return 0; 
+	}
+
+	delay(1);
+  return(1);
+}
+
+//I2C Read 16bit address, 16bit data
+byte ArduCAM::rdSensorReg16_16(uint16_t regID, uint16_t* regDat)
+{
+	uint16_t temp;
+	Wire.beginTransmission(sensor_addr >> 1);
+	Wire.write(regID >> 8);
+	Wire.write(regID & 0x00FF); 	
+	Wire.endTransmission();
+	
+	Wire.requestFrom((sensor_addr >> 1),2);
+	if(Wire.available())
+	{
+		temp = Wire.read();
+		*regDat = (temp << 8) | Wire.read();
+	}
+
+	delay(1);
+  return(1);
+}
+
+//I2C Array Write 8bit address, 8bit data
+int ArduCAM::wrSensorRegs8_8(const struct sensor_reg reglist[])
+{
+	int err = 0;
+	uint16_t reg_addr = 0;
+	uint16_t reg_val = 0;
+	const struct sensor_reg *next = reglist;
+	
+	while ((reg_addr != 0xff) | (reg_val != 0xff))
+	{		
+		reg_addr = pgm_read_word(&next->reg);
+		reg_val = pgm_read_word(&next->val);
+		err = wrSensorReg8_8(reg_addr, reg_val);
+   	next++;
+#if defined(ESP8266)
+	  yield();
```

Comment:
Where is `yield` coming from? Was that code compiled with the ESP8266 directive ?